exchanges/sharedtestvalues/sharedtestvalues.go

```diff
@@ -103,3 +108,45 @@ func SkipTestIfCannotManipulateOrders(t *testing.T, exch exchange.IBotExchange,
 func AreAPICredentialsSet(exch exchange.IBotExchange) bool {
 	return exch.VerifyAPICredentials(exch.GetDefaultCredentials()) == nil
 }
+
+// EmptyStringPotentialPattern is a regular expression pattern for a potential
+// empty string into float64
+var EmptyStringPotentialPattern = `.*float64.*json:"[^"]*,string".*`
+
+// ForceFileStandard will check all files in the current directory for a regular
+// expression pattern. If the pattern is found the test will fail.
+func ForceFileStandard(t *testing.T, pattern, reason string) {
+	t.Helper()
+
+	r := regexp.MustCompile(pattern)
+
+	root := "." // Specify the root directory to start walking from
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		if !info.IsDir() && strings.HasSuffix(path, ".go") {
+			fileContents, err := ioutil.ReadFile(path)
+			if err != nil {
+				t.Fatalf("Failed to read file: %v", err)
+			}
+
+			lines := bytes.Split(fileContents, []byte("\n"))
+			for x, line := range lines {
+				if r.Match(line) {
+					t.Errorf("File: %s line contains pattern [%s] match with [%s] at line %d", path, pattern, string(line), x+1)
+				}
+			}
+		}
+		return nil
+	})
+
+	if t.Failed() {
+		t.Fatalf("Failed due to: %s", reason)
+	}
+
+	if err != nil {
```

Comment:
Err should be first

Reply:
I didn't want to return an error so I can highlight all lines that needed to be fixed in one block. I checked failed first to post reason if there were any found and then checked error afterwards because that would probably an IO error.

Reply:
I appreciated what you did, but it was a bit different. You've even built a way to append errors that would highlight all the issues that arise. But thank you for making those changes